src/lib/y2network/interface.rb

```diff
@@ -45,6 +45,8 @@ class Interface
     attr_reader :configured
     # @return [HwInfo]
     attr_reader :hardware
+    # @return [Symbol] Mechanism to rename the interface (nil -no rename-, :bus_id or :mac)
+    attr_accessor :renaming_mechanism
```

Comment:
this looks quite limiting for me as I would hope that in future we can allow to rename interface more flexible, e.g. by user provided bus_id or by different udev value, which is currently not supported ( and means that for e.g. usb attached wifi you cannot set udev name, as udev rules are empty in dialog ).

Reply:
You are right, I did not though about hotplug devices. Maybe we could add a `renaming_value` (provided by the user). But I would keep the `renaming_mechanism` so we know against which attribute we should match.

Reply:
Or we could even take those `renaming_mechanism/renaming_value` to a separate class.

Reply:
maybe it will be more flexible to have it in own class.